win/rl/pynethack.cc

```diff
@@ -222,11 +227,16 @@ PYBIND11_MODULE(_pynethack, m)
              py::arg("blstats") = py::none(), py::arg("message") = py::none(),
              py::arg("program_state") = py::none(),
              py::arg("internal") = py::none(),
-             py::arg("inv_glyphs") = py::none(), py::keep_alive<1, 2>(),
-             py::keep_alive<1, 3>(), py::keep_alive<1, 4>(),
-             py::keep_alive<1, 5>(), py::keep_alive<1, 6>(),
-             py::keep_alive<1, 7>(), py::keep_alive<1, 8>(),
-             py::keep_alive<1, 9>(), py::keep_alive<1, 10>())
+             py::arg("inv_glyphs") = py::none(),
+             py::arg("inv_letters") = py::none(),
+             py::arg("inv_oclasses") = py::none(),
+
+             py::keep_alive<1, 2>(), py::keep_alive<1, 3>(),
```

Comment:
Can you, please, explain what keep_alive does?

Reply:
See https://pybind11.readthedocs.io/en/stable/advanced/functions.html#keep-alive. Basically, we need to keep track of the lifetimes of objects in Python and C++. Python's objects are refcounted, and this instruction (I believe), adds a refcount to the argument (the numpy arrays we fil things in) that will be decref'd when the `Nethack` object gets destroyed[1]. That's not 100% the right behaviour when `set_buffers` is called several times but seemed easy enough.

[1] Yes, this underlying logic fills in a supplied buffer w/o additional copies for additional performance. That can be dangerous when the user stores return values longer than one step w/o copying, which is why we also have a `copy=True` option.